jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5074,9 +5074,12 @@ public String descriptorString() {
 		}
 /*[ENDIF] JAVA_SPEC_VERSION >= 15 */
 		if (this.isArray()) {
-			return name;
+		} else if (this.isPrimitiveClass()) {
+			name = "Q" + name + ";";
```

Comment:
`StringBuilder` has better perf than String concat (`name = "Q" + name + ";"`). Example of translation:
https://github.com/eclipse/openj9/blob/aa57332f7bbc74aaabbbf10a6c32c41bc6176a16/jcl/src/java.base/share/classes/java/lang/Class.java#L2279-L2280

Reply:
As per https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.18.1,
> An implementation may choose to perform conversion and concatenation in one step to avoid creating and then discarding an intermediate String object. To increase the performance of repeated string concatenation, a Java compiler may use the StringBuffer class or a similar technique to reduce the number of intermediate String objects that are created by evaluation of an expression. 

@tajila Should we leave it as `String concat`?

Reply:
I believe in JDK11 that code sequence will use indy, if you believe that String.concat will yield better perf then we should use that instead.

Reply:
`indy` will lead to bytecode generation for the `MemberName`. `StringBuilder` will avoid bytecode generation and require fewer `String` objects. +1 for `StringBuilder`.